### ISSUE TITLE
brew is forcing use of system ruby and then failing because it is too old 

### ISSUE THREAD

**Issue by djbclark** [Linuxbrew/homebrew-core#5910]
Sometime after #5164 a change was made such that brew is somehow trying to use system ruby instead of what is in PATH. It tries to use system ruby libs even if /usr/bin is not in the PATH at all!

```
/usr/lib/ruby/1.9.1/rubygems/version.rb:191:in `strip!': can't modify frozen String (RuntimeError)
        from /usr/lib/ruby/1.9.1/rubygems/version.rb:191:in `initialize'
        from -e:1:in `new'
        from -e:1:in `<main>'
/tmp/.djbclark/.linuxbrew/Homebrew/Library/Homebrew/brew.rb:12:in `<main>': Homebrew must be run under Ruby 2.3! You're running 1.9.3. (RuntimeError)
```

This occurs both with existing install and when trying to do a new install. The ruby in $PATH is 
`$HOME/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/bin/ruby`

I'm guessing maybe ruby does something weird with libraries such that maybe it is actually ruby 2.3.3 trying to use the system 1.9.1 libraries, as strace doesn't show /usr/bin/ruby actually being executed:

```
$ grep exec strace.log
execve("$HOME/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/bin/ruby", ["ruby", "install"], [/* 56 vars */]) = 0
```

But somehow through FM this still happens:

```
$ grep 1.9.1 strace.log
mmap(NULL, 2129016, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7fb8515f6000
wait4(31023, /usr/lib/ruby/1.9.1/rubygems/version.rb:191:in `strip!': can't modify frozen String (RuntimeError)
        from /usr/lib/ruby/1.9.1/rubygems/version.rb:191:in `initialize'
```

**Comment by iMichka** [comment]
Hi. Thanks for reporting this.

This is a duplicate of https://github.com/Linuxbrew/brew/issues/570

You will find a workaround and some explanations in the issue. Closing here.
